lib/jnpr/junos/utils/sw.py

```diff
@@ -238,6 +242,71 @@ def validate(self, remote_package, issu=False, **kwargs):
             rc, output_msg))
         return 0 == rc
 
+    def _issu_requirement_validation(self):
+        """
+        Checks:
+            * The master Routing Engine and backup Routing Engine must be
+                running the same software version before you can perform a
+                unified ISSU.
+            * Check GRES is enabled
+            * Check NSR is enabled
+            * Check commit synchronize is enabled
+            * Verify that NSR is configured on the master Routing Engine (re0)
+                by using the "show task replication" command.
+            * Verify that GRES is enabled on the backup Routing Engine (re1)
+                by using the show system switchover command.
+
+        :returns:
+            * ``True`` if validation passes.
+            * * ``False`` otherwise
+        """
+        self.log('ISSU requirement validation: The master Routing Engine and\n'
+                 'backup Routing engine must be running the same software\n'
+                 'version before you can perform a unified ISSU.')
+        if not (self._dev.facts['2RE'] and self._dev.facts['version_RE0'] == self._dev.facts['version_RE1']):
+            self.log('Requirement FAILED: The master Routing Engine (%s) and \n'
+                     'backup Routing Engine (%s) must be running the same \n'
+                     'software version before it can perform a unified ISSU' %
+                     (self._dev.facts['version_RE0'],
+                      self._dev.facts['version_RE1']))
+            return False
+        self.log('Checking GRES status')
+        conf = self._dev.rpc.get_config(filter_xml=etree.XML(
+            '<configuration><chassis><redundancy><graceful-switchover/></redundancy></chassis></configuration>'))
```

Comment:
Any time we are checking the config, we need to be very careful. We must account for configuration groups, commit scripts, etc. We also must check the committed configuration rather than the default candidate configuration.

This change would handle committed config, config groups, and commit scripts:
```
    conf = self._dev.rpc.get_config(filter_xml=etree.XML(
        '<configuration><chassis><redundancy><graceful-switchover/></redundancy></chassis></configuration>')
        options={'database': 'committed', 'inherit': 'inherit', 'commit-scripts': 'apply',})
```

However, there are still multiple ways this could fail. This includes ephemeral DB and a lack of permission to view the configuration.

Whenever possible, we should try to check with operational commands rather than configuration checks. In this case, I don't see a way to avoid the configuration check, but we should at least consider there are known possibilities for GRES to be correctly configured, but this check to fail.